ambry-store/src/main/java/com/github/ambry/store/LogSegment.java

```diff
@@ -32,6 +33,7 @@
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.zip.CRC32;
 import net.smacke.jaydio.DirectRandomAccessFile;
+import net.smacke.jaydio.align.DirectIoByteChannelAligner;
```

Comment:
I think it's great that we're investigating using direct I/O 👍 

The https://github.com/smacke/jaydio appears to be abandoned, so do you have any thoughts on how we might fork the repo or otherwise maintain it ourselves in the future?

How will we measure whether or not this implementation is an improvement?

Reply:
That's right, this library hasn't been updated in the last three years, but it should still be working.

It's hard to messure the disk io performance we have some many disk iops from all sources. One thing I can think so is the average write size and the number of write requests. We are using any buffer when performing direct io in compaction, and we have a lot of small records to write to the target log segment. After this PR, we should be able to see number of write requests goes down and the average of write size increases. 

This is a good thing for disk io performance as we don't want that many small writes after all.